### PR TITLE
Add BCR configurations for publishing releases to the central registry

### PR DESCRIPTION
**Pull request by Bencodes**
Adding BCR configurations so that we can start deploying future releases to the Bazel central registry.

This pull request pulls in the `bazel-contrib` Github actions for creating Github releases and publishing to the central registry.

### CODE REVIEW

**Comment by restingbull** [comment]
Added code owners fixes to bring it inline with the maintainers list.

**Review by restingbull — APPROVED** [review]

**Comment by Bencodes** [comment]
> Added code owners fixes to bring it inline with the maintainers list.

Thanks!

File: .bcr/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+matrix:
+  platform: [ "macos", "ubuntu2004" ]
+
+tasks:
+  verify_targets_workspace:
+    name: "Verify build targets"
+    platform: ${{ platform }}
+    build_flags:
+      - "--enable_bzlmod=false"
+    build_targets:
```
Comment (Bencodes) [inline .bcr/presubmit.yml:10]: Worth noting that we don't actually check tests into the release binary of `rules_kotlin` so there isn't much to run on the Bazel central registry presubmit side.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,45 +1,15 @@
...
-      - name: Upload bazel-bin/rules_kotlin_release.tgz.sha256
-        uses: svenstaro/upload-release-action@2.2.1
-        with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: bazel-bin/rules_kotlin_release.tgz.sha256
-          asset_name: rules_kotlin_release.tgz.sha256
-          tag: ${{ github.ref }}
-          overwrite: true
+  release:
+    uses: bazel-contrib/.github/.github/workflows/release_ruleset.yaml@v4
```
Comment (Bencodes) [inline .github/workflows/release.yml:13]: This changes howe we cut releases in `rules_kotlin`. The way to create releases is to now push a tag like `v4.0`

File: .github/workflows/ci.bazelrc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# This file contains Bazel settings to apply on CI only.
+# It is referenced with a --bazelrc option in the call to bazel in ci.yaml
+common --curses=no
+common --enable_bzlmod
```
Comment (Bencodes) [inline .github/workflows/ci.bazelrc:4]: This config is only used by `bazel-contrib/.github/.github/workflows/release_ruleset.yaml@v4`

**Comment by Bencodes** [comment]
Tagged a release from this branch to see if everything works https://github.com/bazelbuild/bazel-central-registry/pull/999

**Pull request merged**